src/core/server/elasticsearch/client/client_wrapper.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { TransportRequestOptions } from '@elastic/elasticsearch/lib/Transport';
+import { Headers } from '../../http/router';
+import { ClientFacade } from './client_facade';
+
+export const getClientFacade = (client: Client, headers: Headers = {}): ClientFacade => {
+  const addHeaders = (options?: TransportRequestOptions): TransportRequestOptions => {
+    if (!options) {
+      return {
+        headers,
+      };
+    }
+    // TODO: do we need to throw in case of duplicates as it was done
+    //       in legacy? - src/core/server/elasticsearch/scoped_cluster_client.ts:L88
+    return {
+      ...options,
+      headers: {
+        ...options.headers,
+        ...headers,
+      },
+    };
```

Comment:
Self explanatory comment: Do we want to reproduce the behavior of 

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/core/server/elasticsearch/scoped_cluster_client.ts#L82-L97

?

Note that this depends on whether we will expose the `TransportRequestOptions` options in our `ClientFacade`

Reply:
I don't think we need this logic. It was an ad-hoc solution introduced in https://github.com/elastic/kibana/pull/34610#discussion_r272610055
And btw it's opposite to the logic in `elasticsearch-js`. I also feel it's more natural when a value from a more specific context (method call in this case) take precedence.